plugins/alerts/frontend/public/javascripts/countly.views.js

```diff
@@ -12,811 +12,483 @@
-        var count = alertsPlugin.getCount();
-        $("#alerts-running-sum").text(count.r);
-        $("#alerts-total-sum").text(count.t);
-        $("#alerts-today-sum").text(count.today);
+    mounted: function () {
     },
-    widgetDrawer: {
-        loadRatingOptions: function(selected) {
-            var ratings = [
-                {value: 1, name: jQuery.i18n.map["star.one-star"]},
-                {value: 2, name: jQuery.i18n.map["star.two-star"]},
-                {value: 3, name: jQuery.i18n.map["star.three-star"]},
-                {value: 4, name: jQuery.i18n.map["star.four-star"]},
-                {value: 5, name: jQuery.i18n.map["star.five-star"]},
-            ];
-            $("#single-target2-dropdown").clySelectSetItems(ratings);
-            if (selected) {
-                $("#single-target2-dropdown").clySelectSetSelection(ratings[selected - 1].value, ratings[selected - 1].name);
+    methods: {
+        appSelected: function (val, notReset) {
+            var self = this;
+            this.$data.apps= [val];
+            if (val) {
+                countlyEvent.getEventsForApps([val], function(eventData) {
+                    const eventOptions = eventData.map(function(e){
+                        return {value: e.value, label: e.name};
+                    });
+                    self.$data.alertDefine['event']["target"] = eventOptions;
+                });
             }
-            else {
-                $("#single-target2-dropdown").clySelectSetSelection("", "Select a rating");
+            if (!notReset) {
+                this.resetAlertCondition();
             }
         },
-        loadAppViewData: function(selectedView) {
-            var appID = $("#single-app-dropdown").clySelectGetSelection();
-            var self = this;
-            self.selectedView = selectedView;
-            if (appID) {
-                alertsPlugin.getViewForApp(appID, function(viewList) {
-                    $("#single-target2-dropdown").clySelectSetItems(viewList);
-                    if (self.selectedView) {
-                        alertsPlugin.getViewForApp(appID, function() {
-                            var selectedViewName = '';
-                            for (var i = 0; i < viewList.length; i++) {
-                                if (viewList[i].value === self.selectedView) {
-                                    selectedViewName = viewList[i].name;
-                                }
-                            }
-                            $("#single-target2-dropdown").clySelectSetSelection(self.selectedView, selectedViewName);
-                        });
-                    }
-                    else {
-                        $("#single-target2-dropdown").clySelectSetSelection("", "Select a View");
-                    }
-                });
+        dataTypeSelected: function (val, notRest) {
+            this.$data.appsSelectorOption = Object.assign([], this.$data.defaultAppsSelectorOption);
```

Comment:
please replace **this.$data** with **this** probably everywhere, unless its absolutely required.